src/server/worker/logs/logger.go

```diff
@@ -268,8 +273,10 @@ func (logger *taggedLogger) Write(p []byte) (_ int, retErr error) {
 // statements to object storage.  Returns a pointer to the generated pfs.Object
 // as well as the total size of all written messages.
 func (logger *taggedLogger) Close() (*pfs.Object, int64, error) {
-	close(logger.msgCh)
+	logger.Lock()
+	defer logger.Unlock()
 	if logger.putObjClient != nil {
+		close(logger.msgCh)
```

Comment:
Just curious about how this code works, why does this go inside the `if` statement? IIUC, `putObjClient` is just for when we want to log to object storage (i.e. for writing logs to the stats commit, I think? Maybe that's wrong though)

Reply:
The way the existing code worked, we used whether `putObjClient` is nil to signal whether Close has been called (the channel has been closed) - if a new log event arrived after Close, we would hit the `logger.putObjClient != nil` on line 222 and skip sending it. 

The race was because we close the channel, but then we block until the error group is finished, then we block until the client is done recieving messages, _then_ we set the client to nil. So there was potentially a period of time where the channel was closed and putObjClient was not nil, which led to a send on a closed channel.

I moved `close(logger.msgCh)` inside the block because otherwise calling Close would close the channel twice, which also panics.